geode-assembly/src/test/java/org/apache/geode/rest/internal/web/RestServersJUnitTest.java

```diff
@@ -15,43 +15,63 @@
 
 package org.apache.geode.rest.internal.web;
 
+import static org.apache.geode.distributed.ConfigurationProperties.HTTP_SERVICE_PORT;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assume.assumeTrue;
 
-import org.apache.geode.distributed.internal.DistributionConfig;
-import org.apache.geode.test.dunit.rules.ServerStarterRule;
-import org.apache.geode.test.junit.categories.IntegrationTest;
-import org.apache.geode.test.junit.categories.RestAPITest;
 import org.apache.http.HttpResponse;
+import org.apache.http.HttpStatus;
 import org.json.JSONArray;
 import org.junit.BeforeClass;
 import org.junit.ClassRule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
 
+import org.apache.geode.distributed.internal.DistributionConfig;
+import org.apache.geode.internal.AvailablePort;
+import org.apache.geode.test.dunit.rules.ServerStarterRule;
+import org.apache.geode.test.junit.categories.IntegrationTest;
+import org.apache.geode.test.junit.categories.RestAPITest;
+
 @Category({IntegrationTest.class, RestAPITest.class})
 public class RestServersJUnitTest {
 
+  private static boolean defaultPortWasAvailable;
+  private static int targetPort;
+
+  static {
+    defaultPortWasAvailable = AvailablePort
+        .isPortAvailable(DistributionConfig.DEFAULT_HTTP_SERVICE_PORT, AvailablePort.SOCKET);
+    targetPort = defaultPortWasAvailable ? DistributionConfig.DEFAULT_HTTP_SERVICE_PORT
```

Comment:
the entire test is to make sure the server's default port works. There is no need to pick a random port when default port is not available (we have tests for that already)

Reply:
If we want to keep this test (which makes sense to me), perhaps we can isolate just this test to a DUnit test with a comment explaining why it has to be that way for CI to run reliably?

Reply:
I don't think it's really worth the extra load of being in the DUnits, especially since it's not really a DUnit.  I think Jinmei just meant, between this comment and the one below, is that I'm trivializing the test by accepting anything other than the default port -- that the server starts on port 7070 when nothing is specified is exactly what this is meant to be testing.

Ignoring the test when the port is occupied is good, I think.  It keeps it correctly categorized but doesn't interrupt testing when the environment is dirty.